### PR TITLE
added automated doctest to decimal_to_hexadecimal.py in conversions

### PR DESCRIPTION
**Pull request by jasper256**

### CODE REVIEW

**Comment by cclauss** [comment]
Please add tests for a negative number (-256) a floating point number (16.16) and a hex number (0xfffff).  Thanks.

**Comment by jasper256** [comment]
Will do @cclauss. I might add an assert statement to weed out floats that don't equal an integer.  I'll try to get it done tomorrow if I have time

**Comment by cclauss** [comment]
I would be OK with a test that fails...
```
>>> decimal_to_hexadecimal(16.16)  # doctest: +ELLIPSIS
        Traceback (most recent call last):
        ...
        KeyError: ...
```

Also to consider...

The line [__decimal, remainder = divmod(decimal, 16)__](https://docs.python.org/3/library/functions.html#divmod) would allow you to eliminate three other lines in this script.

**Comment by jasper256** [comment]
@cclauss thank you for your suggestions.  I believe that I may have seen divmod before but I honestly forgot it existed.  Anyway, I think you will also find that with more extensive test cases and comments on said cases the exact capabilities of the program are more clear now.

**Comment by cclauss** [comment]
My sense is that the output could be clearer a __0xffff__ ranther than just __ffff__.  Your thoughts?

File: conversions/decimal_to_hexadecimal.py [hunk 1]
```diff
@@ -21,23 +21,60 @@
 }
 
 def decimal_to_hexadecimal(decimal):
-    """ take decimal value, return hexadecimal representation as str """
+    """
+        take integer decimal value, return hexadecimal representation as str 
+        >>> decimal_to_hexadecimal(5)
+        '5'
```
Comment (cclauss) [inline conversions/decimal_to_hexadecimal.py]: I believe that you can drop the quotes if you want.

File: conversions/decimal_to_hexadecimal.py [hunk 2]
```diff
@@ -21,23 +21,60 @@
...
+        >>> decimal_to_hexadecimal(999098)
+        'f3eba'
+        >>> # negatives work too
+        >>> decimal_to_hexadecimal(-256)
+        '-100'
+        >>> # floats are acceptable if equivalent to an int
+        >>> decimal_to_hexadecimal(17.0)
+        '11'
+        >>> # other floats will error
+        >>> decimal_to_hexadecimal(16.16)
```
Comment (cclauss) [inline conversions/decimal_to_hexadecimal.py]: Let's simplify to:
```python
 >>> decimal_to_hexadecimal(16.16)  # doctest: +ELLIPSIS
Traceback (most recent call last):
...
AssertionError
```

File: conversions/decimal_to_hexadecimal.py (hunk 1 above)
Comment (jasper256) [reply conversions/decimal_to_hexadecimal.py]: When I drop the quotes, the doctest errors.  This is my first time using doctest so I do not know if there is a workaround.

File: conversions/decimal_to_hexadecimal.py (hunk 2 above)
Comment (jasper256) [reply conversions/decimal_to_hexadecimal.py]: Got it.

**Comment by jasper256** [comment]
> My sense is that the output could be clearer as **0xffff** ranther than just **ffff**. Your thoughts?

As a novice programmer I trust your thoughts. I had never seen this notation before, but I looked it up and it seems commonly accepted so I added it.

File: conversions/decimal_to_hexadecimal.py [hunk 3]
```diff
@@ -22,44 +22,34 @@
...
         >>> decimal_to_hexadecimal(4096)
-        '1000'
+        '0x1000'
         >>> decimal_to_hexadecimal(999098)
-        'f3eba'
+        '0xf3eba'
         >>> # negatives work too
         >>> decimal_to_hexadecimal(-256)
-        '-100'
+        '0x-100'
```
Comment (cclauss) [inline conversions/decimal_to_hexadecimal.py]: Please add a test:
```
>>> decimal_to_hexadecimal(-256) == hex(-256)
True
```

Comment (jasper256) [reply conversions/decimal_to_hexadecimal.py]: Done.  This also made me realize that I was supposed to put '-' before '0x', which I fixed

**Comment by jasper256** [comment]
The pleasure was all mine.  Every time you suggested a feature, I had to look it up and learn how to implement it, which is helping me improve as a programmer.  This was my first time contributing to an open source project, and I look forward to working on more in the future!

**Comment by cclauss** [comment]
Awesome!  Working on open source can be very rewarding when done with an attitude of continuous learning and collaboration.  You are off to a nice start here.  Adding support for floats is still possible...https://docs.python.org/3/library/stdtypes.html#float.hex  ;-)
